src/main/scala/io/iohk/ethereum/mining/Miner.scala

```diff
@@ -159,10 +160,11 @@ class Miner(
 
     (0 to numRounds).toStream.map { n =>
       val nonce = (initNonce + n) % MaxNonce
-      val pow = Ethash.hashimoto(headerHash, nonce.toByteArray, dagSize, dag.apply)
-      (Ethash.checkDifficulty(difficulty, pow), pow, nonce, n)
+      val nonceBytes = ByteUtils.padLeft(ByteString(nonce.toUnsignedByteArray), 8)
```

Comment:
`nonce.toByteArray` (signed) was not what we wanted here. We need unsigned byte array padded to 8 bytes

Reply:
Do you think it might have been more straightforward if `nonce` was a `BigInt` or `UInt256` (here and in `BlockHeader`)? I'm just wondering, it's not a right time for changes like that...

Reply:
I was thinking more about `require(nonce.length == 8)` in BlockHeader, but yeah, it's not good time for it